token-lending/program/tests/modify_reserve_config.rs

```diff
@@ -0,0 +1,380 @@
+            config: TEST_RESERVE_CONFIG,
+            mark_fresh: true,
+            ..AddReserveArgs::default()
+        },
+    );
+
+    // Add a different lending market with a *different* owner
+    let lending_market_pubkey = Pubkey::new_unique();
+    let (lending_market_authority, bump_seed) =
+        Pubkey::find_program_address(&[lending_market_pubkey.as_ref()], &spl_token_lending::id());
+
+    let lending_market_owner = Keypair::new();
+    let oracle_program_id = read_keypair_file("tests/fixtures/oracle_program_id.json")
+        .unwrap()
+        .pubkey();
+
+    test.add_packable_account(
+        lending_market_pubkey,
+        u32::MAX as u64,
+        &LendingMarket::new(InitLendingMarketParams {
+            bump_seed,
+            owner: lending_market_owner.pubkey(),
+            quote_currency: QUOTE_CURRENCY,
+            token_program_id: spl_token::id(),
+            oracle_program_id,
+        }),
+        &spl_token_lending::id(),
+    );
+
+    let other_lending_market = TestLendingMarket {
+        pubkey: lending_market_pubkey,
+        owner: lending_market_owner,
+        authority: lending_market_authority,
+        quote_currency: QUOTE_CURRENCY,
+        oracle_program_id,
+    };
+
+    let (mut banks_client, payer, recent_blockhash) = test.start().await;
+    // Test modify reserve config instruction
+    const OPTIMAL_UTILIZATION_RATE_CHANGE: u8 = 10;
+
+    let new_config = ReserveConfig {
+        optimal_utilization_rate: TEST_RESERVE_CONFIG.optimal_utilization_rate
+            - OPTIMAL_UTILIZATION_RATE_CHANGE,
+        loan_to_value_ratio: 50,
+        liquidation_bonus: 5,
+        liquidation_threshold: 55,
+        min_borrow_rate: 0,
+        optimal_borrow_rate: 4,
+        max_borrow_rate: 30,
+        fees: ReserveFees {
+            borrow_fee_wad: 100_000_000_000,
+            flash_loan_fee_wad: 3_000_000_000_000_000,
+            host_fee_percentage: 20,
+        },
+    };
+
+    let mut transaction = Transaction::new_with_payer(
+        &[modify_reserve_config(
+            spl_token_lending::id(),
+            new_config,
+            sol_test_reserve.pubkey,
+            other_lending_market.pubkey,
+            other_lending_market.owner.pubkey(),
+        )],
+        Some(&payer.pubkey()),
+    );
+
+    transaction.sign(&[&payer, &other_lending_market.owner], recent_blockhash);
+
+    let result = banks_client
+        .process_transaction(transaction)
+        .await
+        .map_err(|e| e.unwrap());
+
+    assert_eq!(
+        result.unwrap_err(),
+        TransactionError::InstructionError(
+            0,
+            InstructionError::Custom(LendingError::InvalidAccountInput as u32)
+        )
+    );
+
+    let reserve_info = sol_test_reserve.get_state(&mut banks_client).await;
+    assert_eq!(
+        reserve_info.config.optimal_utilization_rate,
+        TEST_RESERVE_CONFIG.optimal_utilization_rate
+    );
```

Comment:
nit: full check here also
```suggestion
    assert_eq!(
        reserve_info.config,
        TEST_RESERVE_CONFIG
    );
```